srcStatic/GetCommandLineArguments.cpp

```diff
@@ -9,7 +9,7 @@ std::vector<std::string> GetCommandLineArguments()
 {
 	std::vector<std::string> arguments;
 	int argumentCount;
-	LocalResource<LPWSTR*> commandLineArgs = CommandLineToArgvW(GetCommandLineW(), &argumentCount);
+	LocalResource<LPWSTR[]> commandLineArgs{CommandLineToArgvW(GetCommandLineW(), &argumentCount)};
```

Comment:
Sorry for the long wait on reviewing.

I don't understand exactly what is going on here with the curly braces and the brackets. 

It seemed more intuitive to me before this change as the function returned a pointer to an array of LPWSTR. Now I guess you are creating the array of pointers and storing it in the LocalResource? But the syntax isn't clear to me.

I'm sort of leaning towards leaving this as the raw pointer since I felt like I understood it better. But my lack of understanding isn't necessarily a good reason, plus you put time into updating it which I don't want to throw away...

Reply:
The `CommandLineToArgvW` method returns a dynamic array of strings. This is conceptually a `string[]`. As arrays decay to pointers in C++, we were able to store that into variable typed with a pointer rather than a dynamic array. That would effectively be a `string*`. As the explicit non-decayed array type is a bit more accurate, I chose to express that. Hence the change from `LPWSTR*` to `LPWSTR[]`.

The curly braces are uniform initialization syntax. It could potentially have just used regular parenthesis. It's just calling the constructor, and passing the argument inside the `{}`. It would be the same as: `commandLineArgs(CommandLineToArgvW(GetCommandLineW(), &argumentCount))`